src/EnergyPlus/Coils/CoilCoolingDXCurveFitSpeed.cc

```diff
@@ -343,21 +342,38 @@ void CoilCoolingDXCurveFitSpeed::size(EnergyPlusData &state)
         Psychrometrics::PsyRhoAirFnPbTdbW(DataEnvironment::StdBaroPress, RatedInletAirTemp, RatedInletAirHumRat, RoutineName);
 
     bool PrintFlag = true;
-    std::string CompType = this->object_name;
-    std::string CompName = this->name;
+    std::string CompType = DataHVACGlobals::cAllCoilTypes(DataHVACGlobals::CoilDX_Cooling);
+    std::string CompName = this->parentName;
 
     int SizingMethod = DataHVACGlobals::CoolingAirflowSizing;
-    std::string SizingString = "Rated Air Flow Rate [m3/s]";
-    ReportSizingManager::RequestSizing(state, CompType, CompName, SizingMethod, SizingString, this->evap_air_flow_rate, PrintFlag, RoutineName);
+    std::string preFixString;
+    if (maxSpeeds > 1) preFixString = "Speed " + std::to_string(speedNum + 1) + " ";
+    std::string SizingString = preFixString + "Rated Air Flow Rate [m3/s]";
+    DataSizing::DataConstantUsedForSizing = this->original_input_specs.evaporator_air_flow_fraction / this->active_fraction_of_face_coil_area;
+    Real64 tempSize = this->evap_air_flow_rate;
+    DataSizing::DataIsDXCoil = true;
+    if (this->ratedEvapAirFlowRateIsAutosized) {
+        DataSizing::DataBypassFrac = 1 - this->active_fraction_of_face_coil_area;
+        tempSize = DataSizing::AutoSize;
+    }
+    ReportSizingManager::RequestSizing(state, CompType, CompName, SizingMethod, SizingString, tempSize, PrintFlag, RoutineName);
+    DataSizing::DataBypassFrac = 0;
+    DataSizing::DataIsDXCoil = false;
+    this->evap_air_flow_rate = tempSize;
 
     SizingMethod = DataHVACGlobals::CoolingCapacitySizing;
-    SizingString = "Gross Cooling Capacity [W]";
-    ReportSizingManager::RequestSizing(state, CompType, CompName, SizingMethod, SizingString, this->rated_total_capacity, PrintFlag, RoutineName);
+    SizingString = preFixString + "Gross Cooling Capacity [W]";
+    DataSizing::DataConstantUsedForSizing = this->original_input_specs.gross_rated_total_cooling_capacity_ratio_to_nominal;
+    tempSize = this->rated_total_capacity;
+    if (this->ratedGrossTotalCapIsAutosized) tempSize = DataSizing::AutoSize;
+
+    ReportSizingManager::RequestSizing(state, CompType, CompName, SizingMethod, SizingString, tempSize, PrintFlag, RoutineName);
+    this->rated_total_capacity = tempSize;
```

Comment:
@rraustad Remember this from way back in Feb or so? `DataConstantUsedForSizing` works fine when the coil is in an airloop, but it is not used for sizing when `CurZoneEqNum > 0`. None of our test files use the new DX coil this way, but this unit test does `UnitarySystemModel_MultiSpeedDXCoilsDirectSolutionTest`. What do you suggest as a remedy? I tried setting both `DataConstantUsedForSizing` and `DataFractionUsedForSizing` but that didn't work. This affecxts both the evap air flow rate and capacity for the speed.

Reply:
That will be fixed in the refactored autosizing library which should drop soon. I suggest setting both Data* vars and merge develop when the autosizing branch is merged.

    Real64 CoolingCapacitySizer::size(EnergyPlusData &state, Real64 _originalValue, bool &errorsFound)

    if (this->dataEMSOverrideON) {
        this->autoSizedValue = this->dataEMSOverride;
    } else if (this->dataConstantUsedForSizing >= 0 && this->dataFractionUsedForSizing > 0) {
        // back and forth if dataConstantUsedForSizing should be > or >= 0 to make this work for AutoCalculate
        this->autoSizedValue = this->dataConstantUsedForSizing * this->dataFractionUsedForSizing;
    } else {
        if (this->curZoneEqNum > 0) {


    Real64 CoolingAirFlowSizer::size(EnergyPlusData &EP_UNUSED(state), Real64 _originalValue, bool &errorsFound)

    if (this->dataEMSOverrideON) {
        this->autoSizedValue = this->dataEMSOverride;
    } else if (this->dataConstantUsedForSizing > 0 && this->dataFractionUsedForSizing > 0) {
        this->autoSizedValue = this->dataConstantUsedForSizing * this->dataFractionUsedForSizing;
    } else {
        if (this->curZoneEqNum > 0) {

Reply:
Setting both didn't work. I may just roll back some of these sizing changes so this can get in and let you figure it out in the autosizing branch. I guess we'll see who gets in first.

Reply:
The autosizing branch is stable and could go in at any time.  I will still have API stuff to work on and documentation fixes, but the code itself appears to be good to go now.

Reply:
I wouldn't roll back, it will work once the autosizing branch is merged.